### PR TITLE
Signi nand signup

### PR DESCRIPTION
**Pull request by dswcodes0**
completed log in and sign up functionality

### CODE REVIEW

**Comment by vqdo** [comment]
@dswcodes0 Could you point this to the main branch instead please? 

Screenshot 2023-07-21 at 10 41 09 AM

If you click on "Files Changed" after submitting a PR, you can check what reviewers see which should give you a gut check on whether it was configured as expected.

File: statchatexpress/package.json [hunk 1]
```diff
@@ -14,10 +14,12 @@
...
     "dotenv": "^16.3.1",
     "express": "^4.18.2",
+    "express-session": "^1.17.3",
     "morgan": "^1.10.0",
     "pg": "^8.11.1",
     "pg-hstore": "^2.3.4",
     "sequelize": "^6.32.1",
-    "sequelize-cli": "^6.6.1"
+    "sequelize-cli": "^6.6.1",
+    "uuid": "^9.0.0"
```
Comment (vqdo) [inline statchatexpress/package.json:23]: what's this used for?

File: statchatui/src/SignUp/SignUp.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      const response = await fetch("http://localhost:3002/users", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(formData),
+      });
+
+      if (response.ok) {
+        alert("Sign-up successful!");
```
Comment (vqdo) [inline statchatui/src/SignUp/SignUp.jsx:23]: Let's come back to this and turn these alerts into UI components! Can you create a task for this? (either in your google doc, or in github issues)

File: statchatui/src/SignIn/SignIn.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import "./SignIn.css";
```
Comment (vqdo) [inline statchatui/src/SignIn/SignIn.jsx:3]: It looks like this file doesn't exist. Is that right, or was it not committed?

If it doesn't exist, let's remove this import and all the references to it.

File: statchatui/src/SignIn/SignIn.jsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        alert("login succesful");
+      } else {
+        alert(`Login failed: ${data.message}`);
+      }
+    } catch (error) {
+      console.error("Error submitting form:", error);
+    }
+  };
+  return (
+    <div className="container">
```
Comment (vqdo) [inline statchatui/src/SignIn/SignIn.jsx:33]: Remove these `className` references if the css file doesn't exist

**Review by vqdo — CHANGES_REQUESTED** [review]
Nice work! Please add a test plan so we can see what changes were made.

This should have probably been split into at least 2, potentially 3-4 pull requests :) Submitting smaller PRs will reduce the # of review comments and time spent iterating on them, plus a bunch of other benefits. I would really like to see more frequent, smaller PRs!

File: statchatui/src/SignIn/SignIn.jsx (hunk 4 above)
Comment (dswcodes0) [reply statchatui/src/SignIn/SignIn.jsx:33]: i am going to add css for it later, i will commit the css files but they are blank as of right now

File: statchatui/src/SignIn/SignIn.jsx (hunk 3 above)
Comment (dswcodes0) [reply statchatui/src/SignIn/SignIn.jsx:3]: the file exists for me locally but it is empty, i will see why this is not showing on github

Comment (dswcodes0) [reply statchatui/src/SignIn/SignIn.jsx:3]: the file is on github but it is blank

File: statchatui/src/SignUp/SignUp.jsx (hunk 2 above)
Comment (dswcodes0) [reply statchatui/src/SignUp/SignUp.jsx:23]: okay i will create an issue and come back to this

File: statchatui/src/Landing/Landing.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import "./Landing.css";
+
+const Landing = () => {
```
[the remark this answers is not in the document]
Comment (dswcodes0) [reply statchatui/src/Landing/Landing.jsx:5]: okay i will do smaller pull requests

File: statchatexpress/package.json (hunk 1 above)
Comment (dswcodes0) [reply statchatexpress/package.json:23]: uuid  was to handle user sessions but i didnt get to that yet and i might not be using that.

File: statchatexpress/app.js [hunk 6]
```diff
@@ -21,55 +21,71 @@ app.get('/users', async (req, res) => {
...
+    res.status(500).json({ message: err.message });
+  }
+});
 
-    const statsWithUser = await Stats.findOne({
-      where: { id: stats.id },
-      include: [{ model: User, as: 'user' }]
-    });
+app.post("/users/login", async (req, res) => {
+  console.log(req.body);
```
[the remark this answers is not in the document]
Comment (dswcodes0) [reply statchatexpress/app.js:60]: ok i will get rid of this

File: statchatexpress/app.js [hunk 7]
```diff
@@ -21,55 +21,71 @@ app.get('/users', async (req, res) => {
...
     res.status(500).json({ message: err.message });
   }
 });
 
-// Route to create a new post
-app.post('/stats', async (req, res) => {
+app.post("/users", async (req, res) => {
   try {
-    const stats = await Stats.create(req.body);
+    const newStats = await User.create(req.body);
```
[the remark this answers is not in the document]
Comment (dswcodes0) [reply statchatexpress/app.js:52]: ok

File: statchatexpress/app.js [hunk 8]
```diff
@@ -1,17 +1,17 @@
-import express from 'express';
-import cors from 'cors';
-import morgan from 'morgan';
-import { sequelize } from './database.js';
-import { User, Stats } from './models/index.js';
+import express from "express";
```
[the remark this answers is not in the document]
Comment (dswcodes0) [reply statchatexpress/app.js:1]: okay

**Review by vqdo — APPROVED** [review]
Looks good, thanks for making these changes.

**Pull request merged**